pallets/capacity/src/lib.rs

```diff
@@ -82,8 +85,39 @@ pub mod pallet {
 
 		/// Function that allows a balance to be locked.
 		type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+		/// The minimum required token amount to stake. It facilitates cleaning dust when unstaking.
+		#[pallet::constant]
+		type MinimumStakingAmount: Get<BalanceOf<Self>>;
+
+		/// The maximum number of unlocking chunks a StakingAccountLedger can have. It determines how many concurrent unstaked chunks may exist.
+		#[pallet::constant]
+		type MaxUnlockingChunks: Get<u32>;
 	}
 
+	/// Storage for keeping a ledger of staked token amounts for accounts.
+	#[pallet::storage]
+	#[pallet::getter(fn get_staking_ledger)]
+	pub type StakingAccountLedger<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, StakingAccountDetails<T>>;
+
+	/// Storage to record how many tokens were targeted to an MSA.
+	#[pallet::storage]
+	#[pallet::getter(fn get_target_ledger)]
+	pub type StakingTargetLedger<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		T::AccountId,
+		Twox64Concat,
+		MessageSourceId,
+		StakingTargetDetails<BalanceOf<T>>,
```

Comment:
what happens if there are multiple staking happens from same account to the same MSA? 
My assumption is that probably they add up. 

One thing to consider is that what if the formula that we are using to convert token to capacity changes over time? If that happens then we are going to loose the context of which formula (version) was used for each staking conversion and is it important to track it?